src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -870,7 +871,11 @@ private void mapObservationUnitRows(
 		return allVariables;
 	}
 
-	private void addLocationIdVariable(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
+	void addLocationVariables(final List<MeasurementVariable> environmentDetailAndConditionVariables) {
```

Comment:
@cheryleigh there are two columns for location abbr when it is added to the study, add checking if it is already added.